design/reboot-interface.md

```diff
@@ -89,22 +89,34 @@ section of the BareMetalHost status. This records a time after which the Host
 was last booted using the current image. Processes running prior to this time
 may be assumed to have been stopped.
 
-A new date-time field, ``lastReboot``, will be added to the ``provisioning``
+A new date-time field, ``lastRebootRequest``, will be added to the ``provisioning``
 section of the BareMetalHost status. This records a time after which the Host
-was last requested to reboot.
+was last requested to reboot (because we cannot trust any value from the user,
+who even if well intentioned, may have created the timestamp on a machine that 
+was not synchronised with the cluster or has a different timezone).
 
 Since the user interface requirements are still unclear, we will follow 
 standard practices of using an annotation ( ``host.metal3.io/reboot`` ) to 
-trigger reboots.  
+trigger reboots.  It shall contain the UID of the target Machine so that the
+controller can validate that the associated Machine has not changed since the
+annotation was created.
 
 The actual power management will be performed by the Host controller. This is
 necessary to avoid race conditions by ensuring that the ``Online`` flag and any
 reboot requests are managed in the same place. 
 
-The Host controller reconcile loop will watch for the value in ``host.metal3.io/reboot`` 
-to be greater than the ``lastPoweredOn`` *and* for ``Spec.Online`` set to ``true``. 
-This indicates that a reboot is required and the controller will write the current 
-time to ``lastReboot`` and will begin to power cycle the host.
+If
+* the value in ``host.metal3.io/reboot`` matches the associated Machine, and
```

Comment:
No need to check for this IMO. The existence of any reboot annotation is sufficient.